### PR TITLE
Add hardware_version

### PR DESCRIPTION
**Pull request by ludeeus**

### CODE REVIEW

**Review by milanmeu — APPROVED** [review]
@ludeeus Thanks for the Pull request

This PR works with my Canvas and according to the docs, this also works with Light Panels. I have not been able to verify the other models, but they will most likely have this information as well.

File: aionanoleaf/typing.py [hunk 1]
```diff
@@ -85,6 +85,7 @@ class InfoData(TypedDict):
     serialNo: str
     manufacturer: str
     firmwareVersion: str
+    hardwareVersion: str
```
Comment (ludeeus) [inline aionanoleaf/typing.py:88]: This is not used anywhere, which is why I did not update it

**Comment by ludeeus** [comment]
> This PR works with my Canvas and according to the docs, this also works with Light Panels. I have not been able to verify the other models, but they will most likely have this information as well.

That's fine, I have plans to do another PR which will make it more forgiving later

File: aionanoleaf/typing.py (hunk 1 above)
Comment (milanmeu) [reply aionanoleaf/typing.py:88]: It's used here:
https://github.com/milanmeu/aionanoleaf/blob/c1be410702839f39b702349546da6f1bdd16d9ff/aionanoleaf/nanoleaf.py#L247

Comment (ludeeus) [reply aionanoleaf/typing.py:88]: Nope

Comment (milanmeu) [reply aionanoleaf/typing.py:88]: This is required for mypy.

Comment (milanmeu) [reply aionanoleaf/typing.py:88]: :confused: :question: 
`mypy aionanoleaf` output:
```python3
aionanoleaf/nanoleaf.py:257: error: TypedDict "InfoData" has no key "hardwareVersion"
```

Comment (ludeeus) [reply aionanoleaf/typing.py:88]: No place indicated that you had that as a requirement.
I'll try again later once the repo is ready for contributions.

Comment (milanmeu) [reply aionanoleaf/typing.py:88]: I'm sorry this wasn't clear. Meanwhile I added CI with GitHub Actions for mypy and flake8. I will edit the README for this as well.

**Pull request closed without merging**